rbac/api/cross_access/util.py

```diff
@@ -60,9 +70,9 @@ def get_cross_principal_name(target_org, user_id):
     return f"{target_org}-{user_id}"
 
 
-def create_principal_with_tenant(principal_name, associate_tenant):
+def create_principal_with_tenant(principal_name, associate_tenant, user_id):
     """Create cross-account principal in tenant."""
     cross_account_principal, _ = Principal.objects.get_or_create(
-        username=principal_name, cross_account=True, tenant=associate_tenant
+        username=principal_name, cross_account=True, tenant=associate_tenant, user_id=user_id
```

Comment:
user_id is stored in cross account principal and this principal is used in dual write handler.